api/query_engine.go

```diff
@@ -111,6 +125,30 @@ func alignRequests(reqs []models.Req, aggSettings mdata.AggSettings) ([]models.R
 
 	chosenInterval := options[selected].interval
 
+	// by default we select the lowest res (longest time range) option.
+	selectedHighestRes := len(options) - 1
+	// then we loop over the remaining options trying to find the highest res where the
+	// oldest timestamp is still older than our From ts
+	for i := len(options) - 2; i >= 0; i-- {
+		if oldestTs(options[i].ttl) > reqs[0].From {
+			break
+		}
+		selectedHighestRes = i
+	}
```

Comment:
although we should clarify that this is a different means of calculating stuff that is not in use. otherwise this is confusing. i'll fix